modular_skyrat/master_files/code/modules/mob/living/simple_animal/friendly/dogs.dm

```diff
@@ -155,20 +170,33 @@
 	. = ..(gibbed)
 	if(!.)
 		return FALSE
+
+	UnregisterSignal(src, COMSIG_ATOM_ATTACK_HAND)
+	UnregisterSignal(src, COMSIG_PARENT_ATTACKBY)
+	UnregisterSignal(src, COMSIG_ATOM_HITBY)
+	UnregisterSignal(src, COMSIG_ATOM_EMAG_ACT)
+
 	do_sparks(3, 1, src)
 	var/datum/ai_controller/dog/EN = ai_controller
 	LAZYCLEARLIST(EN.current_behaviors)
 
-/mob/living/simple_animal/pet/dog/corgi/borgi/emag_act(user as mob)
+/mob/living/simple_animal/pet/dog/corgi/borgi/proc/on_emag_act(mob/user)
 	if(!emagged)
 		emagged = 1
-		visible_message(span_warning("[user] swipes a card through [src]."), span_notice("You overload [src]s internal reactor."))
-		addtimer(CALLBACK(src, .proc/explode), 60 SECONDS)
+
+		emote("exclaim")
+		set_light_on(TRUE)
+
+		add_fingerprint(user, TRUE)
+		visible_message(span_boldwarning("[user] swipes a card through [src]!"), span_notice("You overload [src]s internal reactor..."))
+		addtimer(CALLBACK(src, .proc/explode), 50 SECONDS)
 
 /mob/living/simple_animal/pet/dog/corgi/borgi/proc/explode()
-	visible_message(span_warning("[src] makes an odd whining noise."))
-	explosion(get_turf(src), 0, 1, 6, 9, 2, TRUE)
-	death()
+	visible_message(span_bolddanger("[src] makes an odd whining noise!"))
+	do_jitter_animation(30)
+	sleep(10 SECONDS) // Here's your ten second warning
```

Comment:
Don't use sleep(), use one of the timers you can add a callback on, and make that call the rest of this proc (which would be conveniently placed into another proc), since it's a better practice.